libminifi/test/unit/FileSystemRepositoryTests.cpp

```diff
@@ -177,4 +177,46 @@ TEST_CASE("FileSystemRepository removes non-existing resource file from purge li
   REQUIRE(content_repo->getPurgeList().empty());
 }
 
+TEST_CASE("Append Claim") {
+  TestController testController;
+  auto dir = testController.createTempDirectory();
+  auto content_repo = std::make_shared<TestFileSystemRepository>();
+
+  auto configuration = std::make_shared<org::apache::nifi::minifi::Configure>();
+  configuration->set(minifi::Configure::nifi_dbcontent_repository_directory_default, dir.string());
+  REQUIRE(content_repo->initialize(configuration));
+
+
+  const std::string content = "well hello there";
+
+  auto claim = std::make_shared<minifi::ResourceClaim>(content_repo);
+  content_repo->write(*claim)->write(as_bytes(std::span(content)));
+
+  // requesting append before content end fails
+  CHECK(content_repo->append(*claim, 0) == nullptr);
+  auto lock = content_repo->append(*claim, content.length());
+  // trying to append to the end succeeds
+  CHECK(lock != nullptr);
+  // simultaneously trying to append to the same claim fails
+  CHECK(content_repo->append(*claim, content.length()) == nullptr);
```

Comment:
Wouldn't it be better if instead of failing, we copied the contents to a new entry, and then appended the new content there? It doesn't necessarily have to be done in the content repo, the session would be fine, too.

Reply:
the newly modified ContentSession::append does just that, this only checks if we can simultaneously lock for the resource for append